axelor-contract/src/main/java/com/axelor/apps/contract/service/ContractService.java

```diff
@@ -18,22 +18,28 @@
 package com.axelor.apps.contract.service;
 
 import com.axelor.apps.account.db.Invoice;
+import com.axelor.apps.contract.db.ConsumptionLine;
 import com.axelor.apps.contract.db.Contract;
+import com.axelor.apps.contract.db.ContractLine;
 import com.axelor.apps.contract.db.ContractTemplate;
+import com.axelor.apps.contract.db.ContractVersion;
 import com.axelor.exception.AxelorException;
+import com.google.common.collect.Multimap;
 import com.google.inject.persist.Transactional;
 import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
 
 public interface ContractService {
 
   /**
-   * Active the contract
+   * Active the contract at the specific date.
    *
-   * @param contract
-   * @param date
+   * @param contract to active.
+   * @param date to use for active contract.
    */
   @Transactional(rollbackOn = {AxelorException.class, Exception.class})
-  public void activeContract(Contract contract, LocalDate date);
+  void activeContract(Contract contract, LocalDate date);
```

Comment:
No need to specify rollbackOn here
See https://github.com/google/guice/wiki/Transactions